scene/2d/canvas_modulate.cpp

```diff
@@ -70,7 +70,7 @@ void CanvasModulate::_bind_methods() {
 void CanvasModulate::set_color(const Color &p_color) {
 
 	color = p_color;
-	if (is_inside_tree()) {
+	if (is_inside_tree() && is_visible()) {
```

Comment:
This should be `is_visible_in_tree()` to take into account the overall visibility given its parents visibility (`is_visible()` only checks that node's `visible` property).

And then no need for `is_inside_tree()` as it's implied by `is_visible_in_tree()`.

Reply:
Changed.